lib/wifi.toit

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2021 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+import net
+import net.udp
+import net.tcp
+
+import .net.modules.wifi as wifi
+import .net.modules.tcp
+import .net.modules.udp
+import .net.modules.dns
+import .esp32
+
+WIFI_CONNECT_TIMEOUT_  ::= Duration --s=10
+WIFI_DHCP_TIMEOUT_     ::= Duration --s=16
+
+connect --ssid/string?=null --password/string="" -> Wifi:
+  if not ssid:
+    config := image_config or {:}
+    wifi_config := config.get "wifi" --if_absent=: {:}
+    ssid = wifi_config.get "ssid"
+    password = wifi_config.get "password"
+
+  wifi := wifi.Wifi
+  try:
+    wifi.set_ssid ssid password
+    with_timeout WIFI_CONNECT_TIMEOUT_: wifi.connect
+    with_timeout WIFI_DHCP_TIMEOUT_: wifi.get_ip
+    return Wifi wifi
+  finally: | is_exception _ |
+    if is_exception: wifi.close
+
+class Wifi extends net.Interface:
```

Comment:
Having looked a bit more at this, I think the right thing is to make `connect` return a `net.Interface` and to make the `wifi.Wifi` class private (call it `WifiInterface_`).

Also, I think this probably still belongs in `net/`, so you would import it with `import net.wifi` and use it with `wifi.connect`.